src/Controller/DefaultController.php

```diff
@@ -21,20 +22,20 @@ class DefaultController extends AbstractController
     /**
      * The index action.
      *
+     * @param FundingOrgFilter $fundingOrgFilter The funding organization filter utility.
+     *
      * @Route("/", name="pelagos_nas_homepage", condition="'%custom_template%' matches '/nas-grp-base/'")
      *
      * @return Response A Response instance.
      */
-    public function nasIndex()
+    public function nasIndex(FundingOrgFilter $fundingOrgFilter)
     {
         $researchGroups = array();
-        if ($this->isGranted('IS_AUTHENTICATED_FULLY')) {
-            if ($this->isGranted('ROLE_DATA_REPOSITORY_MANAGER')) {
-                $researchGroups = $this->container->get('doctrine')->getRepository(ResearchGroup::class)->findAll();
-            } elseif ($this->tokenStorage->getToken()->getUser() instanceof Account) {
-                $researchGroups = $this->tokenStorage->getToken()->getUser()->getPerson()->getResearchGroups();
-            }
+
+        if ($fundingOrgFilter->isActive()) {
+            $researchGroups = $this->get('doctrine')->getRepository(ResearchGroup::class)->findBy(array('id' => $fundingOrgFilter->getResearchGroupsIdArray() ));
```

Comment:
Maybe we can start using the new ResearchGroupReposity soon ;-)

Reply:
Also, should they be ordered?

Reply:
Not specified in the ticket, but we can do it